### PR TITLE
Implement completions for overrides

### PR DESCRIPTION
**Pull request by kerams**
Implements https://github.com/dotnet/fsharp/issues/6701.

devenv_Rf1fjAc2oo

oNwwTazutx

### CODE REVIEW

File: src/Compiler/Service/FSharpCheckerResults.fs [hunk 1]
```diff
@@ -1008,6 +1008,63 @@ type internal TypeCheckInfo
...
+            |> List.forall (fun x ->
+                if candidate.DisplayName <> x.DisplayName then
+                    true
+                else
+                    match x, candidate with
+                    | FSMeth (valRef = vref1), FSMeth (valRef = vref2) ->
+                        match stripTyEqns g vref1.Type, stripTyEqns g vref2.Type with
+                        | TType_fun (rangeType = ty1), TType_fun (rangeType = ty2) -> not (typeEquiv g ty1 ty2)
+                        | _ -> not (typeEquiv g vref1.Type vref2.Type)
+                    | _ -> false)
```
Comment (kerams) [inline src/Compiler/Service/FSharpCheckerResults.fs:1024]: This is really half-assed, and doesn't work for IL classes, but I could not find a simple way for matching methods by signature. It's a complex topic, so there might not be one.

**Review by psfinaki — APPROVED** [review]
Awesome job. Something I've personally been missing. :)

File: tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.SurfaceArea.netstandard20.debug.bsl
Comment (kerams) [inline tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.SurfaceArea.netstandard20.debug.bsl:1]: I think there might be some inconsistency in how `VerifySurfaceAreaFSharpCompilerService` generates new baselines.

Comment (psfinaki) [reply tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.SurfaceArea.netstandard20.debug.bsl:1]: Got the same impression recently, wonder where can be the problem.

**Pull request merged**